exo/download/hf/hf_shard_download.py

```diff
@@ -77,3 +87,55 @@ async def wrapped_progress_callback(event: RepoProgressEvent):
   @property
   def on_progress(self) -> AsyncCallbackSystem[str, Tuple[Shard, RepoProgressEvent]]:
     return self._on_progress
+
+  async def get_shard_download_status(self) -> Optional[Dict[str, float]]:
```

Comment:
Generally there seems to be a lot of duplication between this and other functions. I think a refactor should be done here.